server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2644,6 +2643,9 @@ private void verifyVmLimits(UserVmVO vmInstance, Map<String, String> details) {
             Map<String, String> customParameters = new HashMap<>();
             customParameters.put(VmDetailConstants.CPU_NUMBER, String.valueOf(newCpu));
             customParameters.put(VmDetailConstants.MEMORY, String.valueOf(newMemory));
+            if (svcOffering.getSpeed() == null) {
```

Comment:
This cond may not be needed, as Unconstrained offering doesn't have CPU speed set and so it's null always. Pls check.

Reply:
@sureshanaparti please check the code. This method is not just for unconstrained offerings. CPU Speed detail is added only for offerings that are dynamic and having null speed

Reply:
> 

Ok, will check @shwstppr